site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+# GEP-724: Refresh Route-Gateway Binding
+
+* Issue URL: [#724](https://github.com/kubernetes-sigs/gateway-api/issues/724)
+* Status: Implementable
+
+## TLDR
+
+This GEP proposes changes to Route-Gateway binding that will result in Routes
+attaching to Gateways with direct references. When supporting Routes in multiple
+namespaces, Gateways will need to specify the namespaces they trust Routes in.
+These changes will slightly simplify the Route-Gateway relationship and make way
+for the future addition of Route inclusion (Routes including other Routes).
+
+## Goals
+
+Refactor cross-namespace Route-Gateway binding to:
+
+* Be more consistent with [cross-namespace references from
+  Routes](/geps/gep-709.md)
+* Provide a clear path to enable Route inclusion (Routes including Routes).
+* Simplify user experience based on initial feedback.
+* Enable other kinds of Route parents in addition to Gateway, this could include:
+    * Routes (as part of Route inclusion)
+    * Custom Gateway resources
+    * Mesh resources
+
+## Out of scope
+
+* Defining how Route inclusion will work.
+
+## Existing Approach
+
+The existing API already supports cross-namespace references. Gateways configure
+the following:
+
+* A Route label selector
+* Namespaces: Same, Selector, or All
+
+Routes then have three options as far as which Gateways can bind to them:
+
+* Same namespace (default)
+* From a list of Gateways
+* All
+
+Although this enables a great deal of flexibility, it can lead to confusion. For
+example, 2 separate label selectors from Gateway can be challenging to compute.
```

Comment:
What does "challenging to compute" mean?  Difficult for humans to reason about? Difficult for an implementation to compute the intersection of routes matching the route selector and routes in namespaces matching the namespace selector?

Reply:
That's a good question, I think my primary concern was user complexity, added some more clarification here.